tests/common/transaction_utils.rs

```diff
@@ -118,3 +131,44 @@ pub fn to_internal_tx(account_tx: &AccountTransaction) -> InternalTransaction {
         max_fee,
     };
 }
+
+fn execute_txs(
+    mut state: CachedState<DictStateReader>,
+    block_context: &BlockContext,
+    txs: Vec<AccountTransaction>,
+) -> (StarknetOsInput, ExecutionHelperWrapper) {
+    let internal_txs: Vec<_> = txs.iter().map(to_internal_tx).collect();
+    let initial_state = copy_state(&state);
+    let execution_infos =
+        txs.into_iter().map(|tx| tx.execute(&mut state, block_context, true, true).unwrap()).collect();
+    os_hints(&block_context, initial_state, internal_txs, execution_infos)
+}
+
+pub fn execute_txs_and_run_os(
+    state: CachedState<DictStateReader>,
+    block_context: BlockContext,
+    txs: Vec<AccountTransaction>,
+) -> Result<CairoPie, SnOsError> {
+    let (os_input, execution_helper) = execute_txs(state, &block_context, txs);
+
+    let result = run_os(
+        config::DEFAULT_COMPILED_OS.to_string(),
+        config::DEFAULT_LAYOUT.to_string(),
+        os_input,
+        block_context,
+        execution_helper,
+    );
+
+    if let Err(ref e) = result {
+        if let Runner(ref r) = e {
+            if let VmException(ref vme) = r {
+                if let Some(traceback) = vme.traceback.as_ref() {
+                    println!("traceback:\n{}", traceback);
+                }
+            }
+        }
+        println!("exception:\n{:#?}", result);
+    }
```

Comment:
This can be simplified to a match statement looking somewhat like this:

```
match &result {
  Err(Runner(VmException(e))) => <traceback>,
  Err(other) => println!(other),
  _ => {},
}
```